test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java

```diff
@@ -18,272 +18,11 @@
-    try (Scanner root = c.createScanner(AccumuloTable.ROOT.tableName(), EMPTY);
-        Scanner meta = c.createScanner(AccumuloTable.METADATA.tableName(), EMPTY)) {
-      root.setRange(TabletsSection.getRange());
-      root.fetchColumnFamily(LogColumnFamily.NAME);
-      TabletColumnFamily.PREV_ROW_COLUMN.fetch(root);
-
-      meta.setRange(TabletsSection.getRange());
-      meta.fetchColumnFamily(LogColumnFamily.NAME);
-      TabletColumnFamily.PREV_ROW_COLUMN.fetch(meta);
-
-      List<String> logs = new ArrayList<>();
-      Iterator<Entry<Key,Value>> both = Iterators.concat(root.iterator(), meta.iterator());
-      while (both.hasNext()) {
-        Entry<Key,Value> entry = both.next();
-        Key key = entry.getKey();
-        if (key.getColumnFamily().equals(LogColumnFamily.NAME)) {
-          logs.add(key.getColumnQualifier().toString());
-        }
-        if (TabletColumnFamily.PREV_ROW_COLUMN.hasColumns(key) && !logs.isEmpty()) {
-          KeyExtent extent = KeyExtent.fromMetaPrevRow(entry);
-          result.put(extent, logs);
-          logs = new ArrayList<>();
-        }
-      }
-    }
-    return result;
-  }
-
-  private final int TIMES_TO_COUNT = 20;
-  private final int PAUSE_BETWEEN_COUNTS = 100;
-
-  private Map<String,WalState> getWALsAndAssertCount(ServerContext c, int expectedCount)
-      throws Exception {
-    // see https://issues.apache.org/jira/browse/ACCUMULO-4110. Sometimes this test counts the logs
-    // before
-    // the new standby log is actually ready. So let's try a few times before failing, returning the
-    // last
-    // wals variable with the the correct count.
-    Map<String,WalState> wals = _getWals(c);
-    if (wals.size() == expectedCount) {
-      return wals;
-    }
-
-    int waitLonger = Wait.getTimeoutFactor(e -> 1); // default to 1
-    for (int i = 1; i <= TIMES_TO_COUNT; i++) {
-      Thread.sleep(i * PAUSE_BETWEEN_COUNTS * waitLonger);
-      wals = _getWals(c);
-      if (wals.size() == expectedCount) {
-        return wals;
-      }
-    }
-
-    fail(
-        "Unable to get the correct number of WALs, expected " + expectedCount + " but got " + wals);
-    return new HashMap<>();
-  }
-
-  static Map<String,WalState> _getWals(ServerContext c) throws Exception {
-    while (true) {
-      try {
-        Map<String,WalState> result = new HashMap<>();
-        WalStateManager wals = new WalStateManager(c);
-        for (Entry<Path,WalState> entry : wals.getAllState().entrySet()) {
-          // WALs are in use if they are not unreferenced
-          result.put(entry.getKey().toString(), entry.getValue());
-        }
-        return result;
-      } catch (WalMarkerException wme) {
-        if (wme.getCause() instanceof NoNodeException) {
-          log.debug("WALs changed while reading, retrying", wme);
-        } else {
-          throw wme;
-        }
-      }
-    }
-  }
+public class WALSunnyDayIT extends WALSunnyDayBaseIT {
```

Comment:
Yeah it is kinda weird and I can not figure out how to improve it.  Its driven by the `@Tag(SUNNY_DAY)`.  Ideally `WALSunnyDayBaseIT` would not exists and ` WALFlakyAmpleIT` would just extend `WALSunnyDayIT`.  However if it did that it would then inherit `@Tag(SUNNY_DAY)` (junit docs says tags are inherited by class but not methods) so it would run as part of the sunny day test which I wanted to avoid.  I can add a comment about the tag thing.